types/jasmine/ts3.1/index.d.ts

```diff
@@ -242,7 +242,7 @@ declare namespace jasmine {
     function createSpyObj<T>(baseName: string, methodNames: SpyObjMethodNames<T>): SpyObj<T>;
 
     function createSpyObj(methodNames: SpyObjMethodNames): any;
-    function createSpyObj<T>(methodNames: SpyObjMethodNames): SpyObj<T>;
+    function createSpyObj<T>(methodNames: SpyObjMethodNames<T>): SpyObj<T>;
```

Comment:
How does this play with #38629

Reply:
That PR needs to also port its changes into the `ts3.1` directory (I left a comment on the PR saying so).

I would assume rebasing once either PR is submitted would be acceptable?

Reply:
This `<T>` change is now reflected in #38629, because I had to update these definitions as well to add `propertyNames?:` after this.

Reply:
This `<T>` change is now reflected in #38629, because I had to update these definitions as well to add `propertyNames?:` after this.